src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinValueInstantiator.kt

```diff
@@ -32,6 +34,9 @@ internal class KotlinValueInstantiator(
 
     private fun List<KTypeProjection>.markedNonNullAt(index: Int) = getOrNull(index)?.type?.isMarkedNullable == false
 
+    private fun SettableBeanProperty.skipNulls(): Boolean =
+        nullIsSameAsDefault || (getAnnotation(JsonSetter::class.java).nulls == Nulls.SKIP)
```

Comment:
@cowtowncoder 
Are there any other patterns to consider regarding `JsonSetter.nulls`?
I understand that all cases other than `SKIP` are handled on the `databind` side.

Reply:
qq: where is `getAnnotation()` defined here? All access should go via `AnnotationIntrospector`, not directly, or via things that collect information through it (I think `BeanDescription` might).

I don't think `ValueInstantiator` really should handle SKIPping; not sure why it is attempted here. In fact I don't think `ValueInstantiator` should handle any of `JsonSetter.Nulls` handling?

Reply:
... although I guess I can see why it is attempted here. Same problem does not apply to Java since no value defaulting is supported.

I also think that `getAnnotation()` comes from `BeanProperty` -- but as per Javadoc, it is meant for accessing custom annotations Jackson does NOT handle; not core annotations. 

Instead, I _think_ you can get this info via `BeanDescription.getMetadata()`; and then `PropertyMetadata.getValueNulls()`.

That way all handling including mix-ins and ConfigOverrides work.

Reply:
Thank you for your suggestion!
You are right, I did not take into account the case where the process is custom.
[11350bc](https://github.com/FasterXML/jackson-module-kotlin/pull/707/commits/11350bca5ca03371e20c0e7c556ed3a63b8540ce)

> Same problem does not apply to Java since no value defaulting is supported.

That is correct.
I had to choose between creating annotation specific to `kotlin-module` or specific support for `Nulls.SKIP`.
I chose the latter because `Nulls.FAIL` was applied even for deserialization by `Creator`.

Reply:
@cowtowncoder 
Does this fix look good?
11350bc

Reply:
Much better @k163377 !

Reply:
Thank you very much!